pluginlib/include/pluginlib/class_loader_imp.hpp

```diff
@@ -446,30 +439,32 @@ std::vector<std::string> ClassLoader<T>::getAllLibraryPathsToTry(
   }
   std::string stripped_library_name_alternative = stripAllButFileFromPath(library_name_alternative);
 
-  // Setup the relative file paths to pair with the search directories above.
-  std::vector<std::string> all_relative_library_paths = {
-    library_name + non_debug_suffix,
-    library_name_alternative + non_debug_suffix,
-    stripped_library_name + non_debug_suffix,
-    stripped_library_name_alternative + non_debug_suffix,
-  };
-  std::vector<std::string> all_relative_debug_library_paths = {
-    library_name + class_loader::systemLibrarySuffix(),
-    library_name_alternative + class_loader::systemLibrarySuffix(),
-    stripped_library_name + class_loader::systemLibrarySuffix(),
-    stripped_library_name_alternative + class_loader::systemLibrarySuffix(),
-  };
-
-  for (auto && current_search_path : all_search_paths) {
-    for (auto && current_library_path : all_relative_library_paths) {
-      all_paths.push_back(current_search_path + path_separator + current_library_path);
-    }
-    // We're in debug mode, try debug libraries as well
-    if (debug_library_suffix) {
+  try {
+    // Setup the relative file paths to pair with the search directories above.
+    std::vector<std::string> all_relative_library_paths = {
+      rcpputils::get_platform_library_name(library_name),
+      rcpputils::get_platform_library_name(library_name_alternative),
+      rcpputils::get_platform_library_name(stripped_library_name),
+      rcpputils::get_platform_library_name(stripped_library_name_alternative)
+    };
+    std::vector<std::string> all_relative_debug_library_paths = {
+      rcpputils::get_platform_library_name(library_name, true),
+      rcpputils::get_platform_library_name(library_name_alternative, true),
+      rcpputils::get_platform_library_name(stripped_library_name, true),
+      rcpputils::get_platform_library_name(stripped_library_name_alternative, true)
+    };
+
+    for (auto && current_search_path : all_search_paths) {
+      for (auto && current_library_path : all_relative_library_paths) {
+        all_paths.push_back(current_search_path + path_separator + current_library_path);
```

Comment:
nitpick: `all_paths` could be pre-allocated with `reserve` (`all_search_paths.size() * all_relative_library_paths * 2`)